From 9fca8e9e489432339ad8078801e026224f7d46ad Mon Sep 17 00:00:00 2001
From: Wang Jie <dave.wang@rock-chips.com>
Date: Mon, 20 Apr 2020 19:59:35 +0800
Subject: [PATCH] arm64: dts: rockchip: modify vcca_33 and vcc_lan for rk3368
 808 evb

solve the machine sleep and wake up, 4G module is abnormal.
Under normal circumstances, sleep and wake up, 4g can not disconnect the network.

Change-Id: Iee28bdcee57747c4f4cbceae8c413b7f46c2b6ad
Signed-off-by: Wang Jie <dave.wang@rock-chips.com>
---
 .../boot/dts/rockchip/rk3368-808-evb.dts      | 20 +++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3368-808-evb.dts b/arch/arm64/boot/dts/rockchip/rk3368-808-evb.dts
index e85962412104..5e7e8c7e4755 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-808-evb.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3368-808-evb.dts
@@ -186,3 +186,23 @@
 
 	};
 };
+
+/*
+ * In sleep mode, should be close vcca_33 and vcc_lan,
+ * but due to small defects in hardware settings and
+ * the system sleeps and wakes up, 4g module can not disconnect the network,
+ * so system sleep mode cannot be turned off vcca_33 and vcc_lan.
+ * This configuration will result in increased power consumption,
+ * please configure according to the actual needs of the project.
+ */
+&vcca_33 {
+	regulator-state-mem {
+		regulator-on-in-suspend;
+	};
+};
+
+&vcc_lan {
+	regulator-state-mem {
+		regulator-on-in-suspend;
+	};
+};
-- 
2.35.3

